tests/network/vmi_infosource.go

```diff
@@ -0,0 +1,101 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kvirtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/libvmi"
+	"kubevirt.io/kubevirt/tests/util"
+)
+
+var _ = SIGDescribe("Infosource", func() {
+	var virtClient kubecli.KubevirtClient
+
+	BeforeEach(func() {
+		var err error
+		virtClient, err = kubecli.GetKubevirtClient()
+		Expect(err).NotTo(HaveOccurred(), "Should successfully initialize an API client")
+
+		tests.BeforeTestCleanup()
+	})
+
+	Context("with a vmi and a bridge interface", func() {
+		var vmi *kvirtv1.VirtualMachineInstance
+
+		const originalMac = "02:00:05:05:05:05"
+		const updatedMac = "02:00:b5:b5:b5:b5"
+
+		BeforeEach(func() {
+			bridgeInterface := libvmi.InterfaceDeviceWithBridgeBinding()
+			vmiSpec := libvmi.NewFedora(
+				libvmi.WithInterface(*libvmi.InterfaceWithMac(&bridgeInterface, originalMac)),
+				libvmi.WithNetwork(kvirtv1.DefaultPodNetwork()))
+			tests.AddUserData(vmiSpec, "cloud-init", "#!/bin/bash\nsudo ip link set dev eth0 address "+updatedMac+"\n")
+
+			var err error
+			vmi, err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmiSpec)
+			Expect(err).NotTo(HaveOccurred())
+			tests.WaitForSuccessfulVMIStart(vmi)
+			tests.WaitAgentConnected(virtClient, vmi)
+		})
+
+		It("should have two entries in vmi status when mac is changed", func() {
+			Eventually(func() error {
+				vmiObj, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				mac, err := getInterfaceMacByInfoSource(vmiObj, "guest-agent")
+				if err != nil {
+					return err
+				}
+				Expect(mac).To(Equal(updatedMac))
+
+				mac, err = getInterfaceMacByInfoSource(vmiObj, "domain")
+				if err != nil {
+					return err
+				}
+				Expect(mac).To(Equal(originalMac))
```

Comment:
I prefer to see an assert on the full list of interfaces reported, not individual ones. This way we will get a full picture if something fails.

Also, if any of the asserts you just added fail, there will be no retry anymore. Was this your intention?

Reply:
will change it to the full list

yes this was the intention
because once the guest-agent appears alone, it means that from this point the other must be domain alone
(it is implicit), even if for example in the beginning there was a non conflict match